pkg/controller.v1beta1/experiment/experiment_controller.go

```diff
@@ -505,3 +511,35 @@ func (r *ReconcileExperiment) ReconcileSuggestions(instance *experimentsv1beta1.
 	}
 	return assignments, nil
 }
+
+func (r *ReconcileExperiment) cleanupTrialResources(instance *experimentsv1beta1.Experiment) error {
+
+	logger := log.WithValues("Experiment", types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()})
+	trials := &trialsv1beta1.TrialList{}
+	labels := map[string]string{consts.LabelExperimentName: instance.Name}
+
+	if err := r.List(context.TODO(), trials, client.InNamespace(instance.Namespace), client.MatchingLabels(labels)); err != nil {
+		logger.Error(err, "Trial List error")
+		return err
+	}
+
+	timeNow := metav1.Now()
+	for i := range trials.Items {
+		if trials.Items[i].IsCompleted() {
+			continue
+		}
+		trials.Items[i].MarkTrialStatusKilled(trialsv1beta1.TrialKilledReason, "experiment already completed")
```

Comment:
What do you mean by Trial Killed status?  Underlying job will be still running ?

Reply:
the trial status will turn to killed， and whether the underlying  job still running or not was decided by the filed `trialSpec.RetainRun`.

code: https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/trial/trial_controller.go#L303

Reply:
Isn't it misleading saying that Trial is killed but underlying job is still running?

Reply:
> Isn't it misleading saying that Trial is killed but underlying job is still running?

I think this is an another question.  Maybe we can create an issue to discuss this case.

my pr was trying to change  trial status when experiment completed.

Reply:
/cc @tenzen-y

Reply:
Hmm, if we want to have Killed status for the Trials, we need to decide how to handle this condition and if it is even useful for the user. Maybe we should just delete Trial in such cases to reduce status complexity.
We don't handle Killed Trial status: https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/trial/trial_controller_util.go#L42.

Reply:
I agree with @andreyvelich.
The killed condition looks curious to me. When the number of failed Jobs reaches `MaxFailedCount`, it might be better to remove running Trial excepted failed Trial.

Reply:
If we remove running Trial in such cases directly, how to update status in experiment? 

in such case, the experiment may be like:
```
status:
  runningTrialList:
  - task-tjgndlgd
  - task-gz78qbcm
  - task-xm6c2lqn
  trials: 6
  trialsRunning: 3
```
how to update the runningTrialList and trialsRunning field?

Reply:
@zhixian82 Maybe we should run [`updateTrialSummary` when Trial is deleted](https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/experiment/util/status_util.go#L56) ?
WDYT @tenzen-y @johnugeorge ?

Reply:
TrialsKilled status is bit confusing term. Are we really Killing the trials or underlying pods? Deletion of running trials can be done ways IMO.  What extra info does it provide to the user?

Reply:
@johnugeorge I don't think we should introduce TrialKilled status.
However, we can delete Trial if that is required from the Experiment controller.
Underlying Trial resources should be deleted also, since they have an `OwnerReference`.

Reply:
> Maybe we should run [updateTrialSummary when Trials are deleted](https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/experiment/util/status_util.go#L56) ?

I agree with that idea. We can update the Experiment status using that helper after unneeded running Trials are deleted.

Reply:
I updated the pr，correct me if i understood wrong。
@andreyvelich @tenzen-y @johnugeorge 

But if we delete running trials directly, the status maybe changed from 
```
status:
  runningTrialList:
  - task-tjgndlgd
  - task-gz78qbcm
  - task-xm6c2lqn
  trials: 6
  trialsRunning: 3
```

to 
```
status:
  ...
  trials: 3
  trialsSucceeded: 3
```
the total trials decreased，this seems unreasonable。

Reply:
That behavior is expected. I think we can remove Job resources, not Trial resources.
@andreyvelich @johnugeorge WDYT?

Reply:
Although, the trial-controller may create new Job resources once we remove Job resources, forcefully.
So, maybe we want to consider the feature of ignoring Trial resources that we don't want to reconcile.

Reply:
> So, maybe we want to consider the feature of ignoring Trial resources that we don't want to reconcile.

Ah, we can mark target Trial resources `Completed`. So we don't need to implement that feature.

Reply:
> But if we delete running trials directly, the status maybe changed from

@zhixian82 @johnugeorge @tenzen-y Any concerns of modifying underlying Experiment status in case of Trial cleanup?

Reply:
I was thinking - We delete underlying jobs and then change Trial status  to TrialKilled status. There is no further reconcile for TrialKilled status and we record the same in experiment status. Any red flags here?

Reply:
@johnugeorge That idea sounds better. @andreyvelich WDYT?

Reply:
Good point. Maybe, we can deal with KIlled Trials similarly to EarlyStopping Trials.